src/viur/core/modules/file.py

```diff
@@ -443,6 +444,34 @@ def write(self, filename: str, content: Any, mimetype: str = "text/plain", width
 
         return skel.toDB()
 
+    def read(self, key: db.Key | int | str | None = None, path: str | None = None) -> io.BytesIO:
+        """
+        Read a file from the Cloudstorage.
+
+        :param key: Key of the LeafSkel that contains the "dlkey" and the "name".
+        :param path: The path of the file in the Cloudstorage Bucket.
+
+
+        :return: Returns the file as a BytesIO buffer.
+        """
+        if not key and not path:
+            raise errors.PreconditionFailed("Please provide a key or a path")
+        if key:
+            skel = self.viewSkel("leaf")
+            if not skel.fromDB(db.keyHelper(key, self.leafSkelCls().kindName)):
+                if not path:
+                    raise errors.NotFound()
+            else:
+                path = f"""{skel["dlkey"]}/source/{skel["name"]}"""
+
+        blob = bucket.blob(path)
+        try:
+            data = blob.download_as_bytes()
+            return io.BytesIO(data)
+        except Exception as e:
+            logging.error(f"Could not read Bolb {e=}")
+            raise errors.NotFound()
```

Comment:
IMHO this function is only used internally, should it really raise `NotFound`?
```suggestion
        data = blob.download_as_bytes()
        return io.BytesIO(data)
```

Reply:
@ArneGudermann what about this proposal?

Reply:
I missed it. Now ok ?